cli/azd/pkg/apphost/generate.go

```diff
@@ -369,6 +430,58 @@ func (b *infraGenerator) LoadManifest(m *Manifest) error {
 				// a part of a server and it should not be created as a separate resource.
 				b.addContainerAppService(name, "postgres")
 			}
+		case "parameter.v0":
+			b.addInputParameter(name, comp.Inputs["value"].Type)
+		case "azure.bicep.v0":
+			if comp.Path == nil {
+				if comp.Parent == nil {
+					return fmt.Errorf("bicep resource %s does not have a path or a parent", name)
+				}
+				// module uses parent
+				continue
+			}
+			if comp.Params == nil {
+				comp.Params = make(map[string]any)
+			}
+			useSecretOutputs := false
+			if comp.ConnectionString != nil && strings.Contains(*comp.ConnectionString, "secretOutputs") {
+				useSecretOutputs = true
+			}
+			keyVaultForParam := ""
+			emptyJsonString := "\"\""
+			for p, pVal := range comp.Params {
+				jsonBytes, err := json.Marshal(pVal)
+				finalParamValue := string(jsonBytes)
+				if err != nil {
+					return fmt.Errorf("marshalling param %s. error: %w", p, err)
+				}
+				if p == "keyVaultName" && useSecretOutputs {
+					keyVaultForParam = name + "kv"
+					keyVaultParam := fmt.Sprintf(
+						"resources.outputs.SERVICE_BINDING_%s_NAME", strings.ToUpper(keyVaultForParam))
+					if finalParamValue == emptyJsonString {
+						finalParamValue = keyVaultParam
+					}
+				}
+				if p == "principalId" && finalParamValue == emptyJsonString {
+					finalParamValue = "guid(resources.outputs.AZURE_CONTAINER_REGISTRY_MANAGED_IDENTITY_ID)"
+				}
+				if p == "principalType" && finalParamValue == emptyJsonString {
+					finalParamValue = "'ServicePrincipal'"
+				}
+				if p == "principalName" && finalParamValue == emptyJsonString {
+					finalParamValue = "resources.outputs.MANAGED_IDENTITY_NAME"
+				}
+				comp.Params[p] = finalParamValue
+			}
+			if useSecretOutputs && keyVaultForParam == "" {
+				return fmt.Errorf(
+					"bicep resource %s has connectionString with secretOutputs but no keyVaultName param", name)
+			}
+			if useSecretOutputs && keyVaultForParam != "" {
+				b.addKeyVault(keyVaultForParam)
+			}
```

Comment:
Nit - could you move most of this code into `addBicep` so this outer loop over resource types is easier to scan.

Reply:
I was trying to avoid making an `add<Resource>()` that returns an error. 
It looks like all other methods where expected to never return an error.  But sure, I can make the exception here.